table/src/main/java/com/bazaarvoice/emodb/table/db/astyanax/AstyanaxTableDAO.java

```diff
@@ -1394,9 +1469,76 @@ public void clearStashTokenRangeSnapshot(String stashId) {
         _stashTableDao.clearTokenRanges(stashId);
     }
 
+    @Timed (name = "bv.emodb.table.AstyanaxTableDAO.listMetadataChangedTables", absolute = true)
+    @Override
+    public Iterator<Map<String, Object>> listMetadataChangedTables(DateTime from, DateTime to, LimitCounter limit) {
+        checkArgument(limit.remaining() > 0, "Limit must be > 0");
+
+        final Iterator<Map<String, Object>> tableIter =
+                _backingStore.scan(_systemTableMetadataChanges, null, limit, ReadConsistency.STRONG);
+
+        // Include only locally available records.
+        // Include only the records which matches the from and to date conditions.
+        Iterators.filter(tableIter, new Predicate<Map<String, Object>>() {
+            @Override
+            public boolean apply(@Nullable Map<String, Object> stringObjectMap) {
+                TableJson tableJson = new TableJson(stringObjectMap);
+                Storage masterStorage = tableJson.getMasterStorage();
+                String masterPlacement = masterStorage.getPlacement();
```

Comment:
Are these even part of the object?  These look like they come from `_systemTable`, but the iterator is iterating over `_systemTableMetadataChanges`, so this shouldn't event work.

Reply:
yeah, I agree. This is incorrect, the above would work with a system table json, but not with the new json.